### ISSUE TITLE
Empty env object in templates

### ISSUE THREAD

**Issue by karthik2804** [fermyon/bartholomew#85]
The `env`  object is missing data about the HTTP request, the path of this resource, and other Spin information in contradiction to the documentation. The only key-value pairs present in it are the environment variables set if any. 

**Steps to reproduce:** 
- Fresh copy of Bartholomew.
- replace docs/templates/main.hbs to just print the `env` object
- run `spin up -env PREVIEW_MODE=1`

**main.hbs:**
```
<ul>
    {{#each env}}
    <li><code>{{@key}}</code>: <code>"{{this}}"</code></li>
    {{/each}}
</ul>
```

**expected result:**
```
{
    PREVIEW_MODE: "1"
    HTTP_REQ: "..."
    PATH: "..."
    SPIN: "..."
}
```

**obtained result:**
```
{
    PREVIEW_MODE: "1"
}
```

The lack of HTTP request and PATH data limit the amount of dynamic generation that can be done. For example, Pagination of content would require path information.

**Comment by radu-matei** [comment]
That information was historically passed into Bartholomew as environment variables because of the CGI spec and Wagi.

Since moving Bartholomew to be a Spin component, most of that information is now passed as headers — see https://spin.fermyon.dev/http-trigger/

**Comment by karthik2804** [comment]
Thanks for pointing to the resource. 

As a follow-up, I see a `request` object which is currently passed as an empty object to `TemplateContext`, would this be the right place to add the information?

If so what are the fields that would need to be added to the `request` object?  Based on the default headers that are passed from spin, only the `spin-full-path` would be needed as the rest can be obtained from the full path. What others would be suggested to be implemented. I can work on submitting a PR once the implementation is discussed. 

Once the headers are accessible from the templates, pagination will be possible using the URL query parameters.

Any inputs would be greatly appreciated!

**Comment by karthik2804** [comment]
Fixed by #90
